Build error :

**Desktop :**
 - OS:  Windows10
 - Filament :  filament-1.9.9
 - IDE:  vs2019 community
 - Backend: [OpenGL]

**Describe the bug:**   
I have fixed some **goto command skip variable initialization errors**  in PlatformWGL.cpp 。

**But  I can't fix below outputs error.  Please help me, Thanks!** :


16>E:\study\graphcs\ar\v1.9.9\filament-1.9.9\libs\utils\include\utils/SingleInstanceComponentManager.h(306,1): warning C4267: 'return': conversion from 'size_t' to 'utils::SingleInstanceComponentManager<filament::Box,uint8_t,filament::math::float4,filament::FRenderableManager::Visibility,utils::Slice<filament::FRenderPrimitive,uint32_t>,std::unique_ptr<filament::FRenderableManager::Bones,std::default_delete<filament::FRenderableManager::Bones>>>::Instance', possible loss of data
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\algorithm(7419,1): **`error C2678: binary '=': no operator found which takes a left-hand operand of type 'const std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>' (or there is no acceptable conversion)`**
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\utility(271,11): message : could be 'std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>::operator =(volatile const std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &)'
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\utility(288,24): message : or       'std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>::operator =<std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>,0>(std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &&) noexcept(<expr>)'
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\utility(277,24): message : or       'std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>::operator =<std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>,0>(const std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float> &) noexcept(<expr>)'
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\algorithm(7419,1): message : while trying to match the argument list '(const std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>, std::pair<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Structure,float>)'
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\algorithm(7541): message : see reference to function template instantiation '_BidIt std::_Insertion_sort_unchecked<_RanIt,_Pr>(const _BidIt,const _BidIt,_Pr)' being compiled
16>        with
16>        [
16>            _BidIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo> *>,float *>,
16>            _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo> *>,float *>,
16>            _Pr=filament::FScene::prepareDynamicLights::<lambda_1>
16>        ]
16>E:\application\vs2019\VC\Tools\MSVC\14.28.29333\include\algorithm(7571): message : see reference to function template instantiation 'void std::_Sort_unchecked<_RanIt,_Fn>(_RanIt,_RanIt,__int64,_Pr)' being compiled
16>        with
16>        [
16>            _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo> *>,float *>,
16>            _Fn=filament::FScene::prepareDynamicLights::<lambda_1>,
16>            _Pr=filament::FScene::prepareDynamicLights::<lambda_1>
16>        ]
16>E:\study\graphcs\ar\v1.9.9\filament-1.9.9\filament\src\Scene.cpp(278): message : see reference to function template instantiation 'void std::sort<utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo> *>,float *>,filament::FScene::prepareDynamicLights::<lambda_1>>(const _RanIt,const _RanIt,_Pr)' being compiled
16>        with
16>        [
16>            _RanIt=utils::Zip2Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo>::Iterator<utils::StructureOfArraysBase<utils::Arena<utils::HeapAllocator,utils::LockingPolicy::NoLock,utils::TrackingPolicy::Untracked>,filament::math::float4,filament::math::float3,filament::FLightManager::Instance,filament::Culler::result_type,filament::math::float2,filament::FScene::ShadowInfo> *>,float *>,
16>            _Pr=filament::FScene::prepareDynamicLights::<lambda_1>
16>        ]
16>Done building project "filament.vcxproj" -- FAILED.
========== Build: 15 succeeded, 1 failed, 35 up-to-date, 0 skipped ==========

--- a/libs/utils/include/utils/StructureOfArrays.h
+++ b/libs/utils/include/utils/StructureOfArrays.h
@@ -181,7 +181,7 @@ public:
         Iterator(Iterator const& rhs) noexcept = default;
         Iterator& operator=(Iterator const& rhs) = default;
 
-        const reference operator*() const { return { soa, index }; }
+        reference operator*() const { return { soa, index }; }
         reference operator*() { return { soa, index }; }
         reference operator[](size_t n) { return *(*this + n); }
 

--- a/libs/utils/include/utils/Zip2Iterator.h
+++ b/libs/utils/include/utils/Zip2Iterator.h
@@ -60,8 +60,11 @@ public:
     Zip2Iterator(Zip2Iterator const& rhs) noexcept = default;
     Zip2Iterator& operator=(Zip2Iterator const& rhs) = default;
 
-    const value_type operator*() const { return { *mIt.first, *mIt.second }; }
-          reference  operator*()       { return { *mIt.first, *mIt.second }; }
+    // MSVC's implementation of std::sort performs
+    //   *iterator = ...
+    // so the return value for operator* overloads must be non-const and assignable.
+    value_type operator*() const { return { *mIt.first, *mIt.second }; }
+    reference  operator*()       { return { *mIt.first, *mIt.second }; }
 
     const value_type operator[](size_t n) const { return *(*this + n); }
           reference  operator[](size_t n)       { return *(*this + n); }

--- a/libs/utils/include/utils/Zip2Iterator.h
+++ b/libs/utils/include/utils/Zip2Iterator.h
@@ -60,9 +60,6 @@ public:
     Zip2Iterator(Zip2Iterator const& rhs) noexcept = default;
     Zip2Iterator& operator=(Zip2Iterator const& rhs) = default;
 
-    // MSVC's implementation of std::sort performs
-    //   *iterator = ...
-    // so the return value for operator* overloads must be non-const and assignable.
     value_type operator*() const { return { *mIt.first, *mIt.second }; }
     reference  operator*()       { return { *mIt.first, *mIt.second }; }
 

